lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -308,8 +308,11 @@ struct ConvertLayoutOpUsingLinearLayoutsConversion
       // Case 3. Transfer between values in the same warp, in which case we try
       //         to move values using warp shuffles, though if the pattern is
       //         complicated enough we may fall back to using shared memory
-      // TODO(Keren): implement warp shuffle instead of using the general
-      // approach that uses shared memory
+      if (auto decomposedCvt =
+              getWarpLayoutConvertDecomposition(srcTy, dstTy)) {
+        transferWithinWarp(op, *decomposedCvt, adaptor, rewriter);
+        return success();
+      }
       return transferWithinBlock(op, srcLayout, dstLayout, adaptor, rewriter);
```

Comment:
Random thought. Since data is only transferred within a warp, you could also try still using smem but changing `barrier` to `bar.warp.sync` between writing and reading. This should improve latency when the warps issue the barrier on different cycles. (Though we would still need a barrier between `ConvertLayoutOp`s).

Reply:
Sure, we can definitely have a test on it.